core/src/main/java/com/graphhopper/reader/osm/WayToEdgeConverter.java

```diff
@@ -0,0 +1,221 @@
+        candidates.add(listFromIterator(edgesByWay.apply(fromWay)));
+        // todonow: no it is not as easy as this, for multiple via ways we also need to find the order of the via ways...
+        viaWays.forEach((LongProcedure) c -> candidates.add(listFromIterator(edgesByWay.apply(c))));
+        candidates.add(listFromIterator(edgesByWay.apply(toWay)));
+
+        // the search starts at *every* edge at level 0
+        for (IntCursor from : candidates.get(0)) {
+            EdgeIteratorState edge = baseGraph.getEdgeIteratorState(from.value, Integer.MIN_VALUE);
+            explore(candidates, edge.getBaseNode(), 1, IntArrayList.from(edge.getEdge(), edge.getBaseNode()), solutions);
+            explore(candidates, edge.getAdjNode(), 1, IntArrayList.from(edge.getEdge(), edge.getAdjNode()), solutions);
+        }
+    }
+
+    private void explore(List<IntArrayList> candidates, int node, int level, IntArrayList curr, List<IntArrayList> solutions) {
+        if (level == candidates.size() - 1) {
+            for (IntCursor to : candidates.get(level)) {
+                if (baseGraph.isAdjacentToNode(to.value, node)) {
+                    IntArrayList solution = new IntArrayList(curr);
+                    solution.add(to.value);
+                    solutions.add(solution);
+                }
+            }
+            return;
+        }
+        for (IntCursor c : candidates.get(level)) {
+            if (baseGraph.isAdjacentToNode(c.value, node)) {
+                int otherNode = baseGraph.getOtherNode(c.value, node);
+                curr.add(c.value, otherNode);
+                explore(candidates, otherNode, level + 1, curr, solutions);
+                // backtrack
+                curr.elementsCount -= 2;
+            }
+        }
+    }
+
+    private static IntArrayList listFromIterator(Iterator<IntCursor> iterator) {
+        IntArrayList result = new IntArrayList();
+        iterator.forEachRemaining(c -> result.add(c.value));
+        return result;
+    }
+
+    public static class EdgeResult {
+        private final IntArrayList fromEdges;
+        private final IntArrayList viaEdges;
+        private final IntArrayList toEdges;
+        private final IntArrayList nodes;
+
+        public EdgeResult(int numFrom, int numVia, int numTo) {
+            fromEdges = new IntArrayList(numFrom);
+            viaEdges = new IntArrayList(numVia);
+            toEdges = new IntArrayList(numTo);
+            nodes = new IntArrayList(numVia + 1);
+        }
+
+        public IntArrayList getFromEdges() {
+            return fromEdges;
+        }
+
+        public IntArrayList getViaEdges() {
+            return viaEdges;
+        }
+
+        public IntArrayList getToEdges() {
+            return toEdges;
+        }
+
+        /**
+         * All the intermediate nodes, i.e. for an edge chain like this:
+         * 0   1   2   3
+         * 0---1---2---3---4
+         * where 0 is the from-edge and 3 is the to-edge this will be [1,2,3]
```

Comment:
do you mean something like this?
```suggestion
         *   a   b   c   d
         * 0---1---2---3---4
         * where a is the from-edge and d is the to-edge this will be [1,2,3]
```

Reply:
Yes, the code-formatting broke my little graphic here (the first line was shifted to the right before)